src/main/java/com/infomaniak/lib/core/networking/NetworkExtensions.kt

```diff
@@ -18,66 +18,21 @@
 package com.infomaniak.lib.core.networking
 
 import android.content.Context
-import android.content.Context.CONNECTIVITY_SERVICE
 import android.net.ConnectivityManager
 import android.net.Network
 import android.net.NetworkCapabilities
 import android.net.NetworkRequest
 import android.os.Build
-import android.util.Log
-import androidx.lifecycle.LiveData
-import io.sentry.Sentry
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.launch
 
-class NetworkExtensions(context: Context) : LiveData<Boolean>() {
+fun Context.networkStatusFlow(): Flow<Boolean> {
+    val networks = mutableSetOf<Network>()
```

Comment:
We were previously using a list instead of a set to store the networks. Did the management of networks change or is there a gain to using a set to store the networks over a list?

Reply:
We are receiving network objects from callbacks, and adding/removing them to our `networks` list.
I proposed that we use a Set instead, to avoid having duplicates.

Reply:
If we can have duplicated networks instances this would let us remove these extra ones and it would make sens to use a Set. But if we use a Set when there are no avantages to using one, then I think it will be misleading

Reply:
@KevinBoulongne 🥊  @LunarX